rust/datafusion/src/physical_plan/mod.rs

```diff
@@ -107,6 +107,13 @@ pub async fn collect(plan: Arc<dyn ExecutionPlan>) -> Result<Vec<RecordBatch>> {
 /// Partitioning schemes supported by operators.
 #[derive(Debug, Clone)]
 pub enum Partitioning {
+    /// Allocate batches using a round-robin algorithm
+    RoundRobinBatch(usize),
+    /// Allocate rows using a round-robin algorithm. This provides finer-grained partitioning
+    /// than `RoundRobinBatch` but also has much more overhead.
+    RoundRobinRow(usize),
```

Comment:
Not that you asked, but if I had to pick 2 of these three schemes to implement, I would pick `RoundRobinBatch` and `Hash` and leave `RoundRobinRow` until later 

The rationale being that I theorize `RoundRobinRow` usecase is much less common (e.g. maybe re-evening output of joins or filters, but I would expect most operators to respect the requested batch size if possible when creating their output)

Reply:
Thanks. That makes sense and I have removed `RoundRobinRow` now.

Reply:
Agree. I was trying to implement the `RoundRobinRow` functionality independently and was going down a route similar to the `StructBuilder` vector of builders route: https://github.com/apache/arrow/blob/master/rust/arrow/src/array/builder.rs#L1600. Staying at the `RecordBatch` level is much more sensible.